test/libcommonserver/log/testlog.cpp

```diff
@@ -45,4 +52,257 @@ void TestLog::testLog() {
+        }
+        logFile.close();
+
+        // compress the log file
+        ExitCause cause = ExitCauseUnknown;
+        ExitCode exitCode = Log::instance()->compressLogFiles(tempDir.path, cause);
+        CPPUNIT_ASSERT_EQUAL(ExitCauseUnknown, cause);
+        CPPUNIT_ASSERT_EQUAL(ExitCodeOk, exitCode);
+
+        // copy the compressed log file to the log directory
+        IoError err = IoErrorSuccess;
+        CPPUNIT_ASSERT_EQUAL(true, IoHelper::copyFileOrDirectory(tempDir.path / "test.log.gz", logDir / "test.log.gz", err));
+        CPPUNIT_ASSERT_EQUAL(IoErrorSuccess, err);
+
+        IoHelper::deleteDirectory(tempDir.path / "test.log.gz", err);
+
+        exitCode = Log::instance()->copyLogsTo(tempDir.path, false, cause);
+        IoHelper::deleteDirectory(logDir / "test.log.gz", err);
+
+        CPPUNIT_ASSERT_EQUAL(ExitCauseUnknown, cause);
+        CPPUNIT_ASSERT_EQUAL(ExitCodeOk, exitCode);
+
+        // Check that `test.log.gz` does not exist anymore.
+        bool exists = false;
+        CPPUNIT_ASSERT_EQUAL(true, IoHelper::checkIfPathExists(tempDir.path / "test.log.gz", exists, err));
+        CPPUNIT_ASSERT_EQUAL(IoErrorNoSuchFileOrDirectory, err);
+        CPPUNIT_ASSERT_EQUAL(false, exists);
+    }
+}
+
+void TestLog::testCopyParmsDbTo(void) {
+    {
+        if (!_parmsDbFileExist()) {
+            std::cout << std::endl << "No .parms.db file, this test will not be relevant (skipped)." << std::endl;
+            LOG_WARN(_logger, "No .parms.db file, this test will not be relevant (skipped).");
+            return;
+        }
+
+        TemporaryDirectory tempDir;
+        const SyncPath parmsDbName = ".parms.db";
+        const SyncPath parmsDbPath = CommonUtility::getAppSupportDir() / parmsDbName;
+
+        uint64_t parmsDbSize = 0;
+        IoError err = IoErrorSuccess;
+        IoHelper::getFileSize(parmsDbPath, parmsDbSize, err);
+
+        
+
+        CPPUNIT_ASSERT_EQUAL(IoErrorSuccess, err);
+        CPPUNIT_ASSERT(parmsDbSize >= 0);
+
+        ExitCause cause = ExitCauseUnknown;
+        ExitCode exitCode = Log::instance()->copyParmsDbTo(tempDir.path, cause);
+        CPPUNIT_ASSERT_EQUAL(ExitCauseUnknown, cause);
+        CPPUNIT_ASSERT_EQUAL(ExitCodeOk, exitCode);
+
+        uint64_t tempDirSize = 0;
+        bool tooDeep = false;
+        IoHelper::getDirectorySize(tempDir.path, tempDirSize, err, 0);
+        CPPUNIT_ASSERT(err == IoErrorSuccess || err == IoErrorMaxDepthExceeded);
+        CPPUNIT_ASSERT_EQUAL(parmsDbSize, tempDirSize);
+    }
+}
+
+void TestLog::testCompressLogs(void) {
+    {
+        TemporaryDirectory tempDir;
+
+        std::ofstream logFile(tempDir.path / "test.log");
+        for (int i = 0; i < 10000; i++) {
+            logFile << "Test log line " << i << std::endl;
+        }
+        logFile.close();
+
+        const SyncPath logDir = tempDir.path / "log";
+        IoError err = IoErrorSuccess;
+        CPPUNIT_ASSERT_EQUAL(true, IoHelper::createDirectory(logDir, err));
```

Comment:
```suggestion
        CPPUNIT_ASSERT(IoHelper::createDirectory(logDir, err));
```

Reply:
Done in commit: https://github.com/Infomaniak/desktop-kDrive/pull/100/commits/96fbf4b3b61e6e7e97cf2f460b24e88cee8b44a4